lib/circuits_sim/gpio/backend.ex

```diff
@@ -42,7 +50,6 @@ defmodule CircuitsSim.GPIO.Backend do
   defp set_pull_mode(_gpio_spec, :not_set), do: :ok
   defp set_pull_mode(gpio_spec, pull_mode), do: GPIOServer.set_pull_mode(gpio_spec, pull_mode)
 
-  defp set_initial_value(_gpio_spec, :not_set), do: :ok
```

Comment:
I removed this unused function, initial_value is 0 or 1.

Reply:
It should be possible to use this if you want to open a GPIO as an output, but you don't want to change its value immediately.

Reply:
Just a thought - since `set_initial_value/1` is a private function, how about adding a first argument to it for whether the gpio is an input or output. If it's an input, then it can skip calling `GPIOServer.write/2`:

```elixir
   defp set_initial_value(:output, gpio_spec, value) when is_integer(value), do: GPIOServer.write(gpio_spec, value)
   defp set_initial_value(_direction, _gpio_spec, _value), do: :ok
```

Reply:
Oh, it makes sense and don't need to separate. How about `set_pull_mode`? Is that func is only for input, right?

Reply:
> It should be possible to use this if you want to open a GPIO as an output, but you don't want to change its value immediately.

I just thought `:not_set` is for `:pull_mode`, no? ref https://hexdocs.pm/circuits_gpio/Circuits.GPIO.html#open/3

Reply:
You're right. It used to be possible to avoid updating the GPIO when opening it for output, but that no longer works with Circuits.GPIO 2.0.

Reply:
I see. If you'd like to fix this on your own (It'll be fast without conversation), I really don't care. Please let me know. Thank you.

Reply:
Sorry for the delay. Your comment fell to the second page of my GitHub notifications and I missed it. I updated your commit. Since you identified the issue, found the fix, and explained it to me when I was mistaken about a Circuits 1.0 feature, I kept your name on the commit. Hope that's ok.